src/server/pkg/worker/api_server.go

```diff
@@ -490,6 +490,9 @@ func HashDatum(pipelineInfo *pps.PipelineInfo, data []*Input) (string, error) {
 		hash.Write(datum.FileInfo.Hash)
 	}
 
+	// We set env to nil because if env contains more than one elements,
+	// since it's a map, the output of Marshal() can be non-deterministic.
+	pipelineInfo.Transform.Env = nil
```

Comment:
You probably need to save the `Env` value here and reset it after  you `Marshal` otherwise when this function returns `Env` will be `nil` and the `Env` vars won't get passed at all.